tests/module.rs

```diff
@@ -0,0 +1,102 @@
+pub mod inner {
+    use snafu::Snafu;
+
+    #[derive(Debug)]
+    pub struct Dummy0;
+
+    #[derive(Debug)]
+    pub struct Dummy1;
```

Comment:
Since the variants all have the same name, I wanted to make sure the generated selectors had the correct type in them. Probably overkill though, so I can remove them all if you like.

Reply:
Oh god, it's half of my motivation for this PR. I'll have a constructor that can fail in `A`, `B`, `C` ways, then the rest of the methods can fail in `A`, `D`, `E`, `F` ways, and the `A` context struct will conflict.